File: ProyectoCivitasJava/src/civitas/CasillaSorpresa.java
/**
 * Proyecto Civitas
 * Sergio Hervás Cobo
 * Javier Suárez Lloréns
 * Programación y Diseño Orientado a Objetos
 * 2º B3 - Grado en Ingeniería Informática
 * E.T.S.I. Informática y de Telecomunicación
 * Universidad de Granada
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author sergio
 */
public class CasillaSorpresa extends Casilla {
    private MazoSorpresas mazo;
    private Sorpresa sorpresa;
    
    CasillaSorpresa(MazoSorpresas mazo, String nombre){
        super(nombre);
        this.mazo = mazo;
    }
    

    @Override
    void recibeJugador(int iactual, ArrayList <Jugador> todos){
        if(jugadorCorrecto(iactual, todos)){
            sorpresa = mazo.siguiente();
            informe(iactual, todos);
            sorpresa.aplicarAJugador(iactual, todos);
        }
    }
    
    @Override
    public String toString(){
        String salida = "Casilla " + getNombre();
        
        
        return salida;
    }
}

File: ProyectoCivitasRuby/lib/modelo/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaSorpresa < Casilla
    def initialize(mazo, nombre)
      super(nombre)
      @mazo = mazo
      @sorpresa = nil
    end
  
    #@Override
    def recibe_jugador(iactual, todos)
      if(jugador_correcto(iactual, todos))
        @sorpresa = @mazo.siguiente()
        informe(iactual, todos)
        @sorpresa.aplicarajugador(iactual, todos)
      end
    end

    #@Override    
    def to_s
      salida = "Casilla #{@nombre}"
    end
    
  end
end